pkg/cliplugins/bind/plugin/bind_compute.go

```diff
@@ -234,21 +236,43 @@ func (o *BindComputeOptions) applyAPIBinding(ctx context.Context, client kcpclie
 	}
 
 	existingAPIExports := sets.NewString()
+	var bindingNamesToPatch []string
 	for _, binding := range apiBindings.Items {
 		if binding.Spec.Reference.Workspace == nil {
 			continue
 		}
-		existingAPIExports.Insert(fmt.Sprintf("%s:%s", binding.Spec.Reference.Workspace.Path, binding.Spec.Reference.Workspace.ExportName))
+		apiExportKey := fmt.Sprintf("%s:%s", binding.Spec.Reference.Workspace.Path, binding.Spec.Reference.Workspace.ExportName)
+		existingAPIExports.Insert(apiExportKey)
+		if desiredAPIExports.Has(apiExportKey) {
+			if len(binding.Labels) == 0 || len(binding.Labels[apisv1alpha1.LabelAPIBindingKindKey]) == 0 {
+				bindingNamesToPatch = append(bindingNamesToPatch, binding.Name)
+			}
+		}
 	}
 
-	diff := desiredAPIExports.Difference(existingAPIExports)
 	var errs []error
+	// patch apibindings with LabelAPIBindingKindKey
+	patchBytes := fmt.Sprintf(`{"metadata":{"labels":{%q:%q}}}`, apisv1alpha1.LabelAPIBindingKindKey, workloadv1alpha1.LabelKindAPIBindingCompute)
+	for _, bindingName := range bindingNamesToPatch {
+		_, err = client.ApisV1alpha1().APIBindings().Patch(ctx, bindingName, types.MergePatchType, []byte(patchBytes), metav1.PatchOptions{})
+		if err != nil {
+			errs = append(errs, err)
+		}
+
+		_, err = fmt.Fprintf(o.Out, "apibinding %s patched to set label %s=%s.\n", bindingName, apisv1alpha1.LabelAPIBindingKindKey, workloadv1alpha1.LabelKindAPIBindingCompute)
+		if err != nil {
+			errs = append(errs, err)
+		}
+	}
```

Comment:
this confuses me. Either the kind label is API and then the controller should copy it from export to binding. Or it is something for the bind command. I would be worried about the latter. Not every bindings comes from the bind command.

Reply:
It expresses the intent to use this binding as an `APIBinding` for workloads (== to be synced).

You could `bind apiexport` a workspace to the `root:compute` `kubernetes` `APIExport`, without any intent to sync related resources to a `SyncTarget`, possibly having some controller reconcile related resources and use them in any other way.

But when you `bind compute`, you clearly express the intent to sync to a `SyncTarget`, and so the `APIBinding`s bound to involved `APIExport`s (those supported by the `SyncTarget`s) would be marked with this workload kind.

Reply:
> Not every bindings comes from the bind command.

I don't think we can demand from the user that only the bind command copies this. It *must* come through a controller.

Reply:
changed to use controller to sync an annotation from apiexport. please review if the annotation name makes sense